src/HDInsight/HDInsight/ManagementCommands/SetAzureHDInsightGatewayCredentialCommand.cs

```diff
@@ -0,0 +1,139 @@
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.HDInsight.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.WindowsAzure.Commands.Common;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.HDInsight
+{
+    [Cmdlet(VerbsCommon.Set, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "HDInsightGatewayCredential", DefaultParameterSetName = SetByNameParameterSet, SupportsShouldProcess = true), OutputType(typeof(HttpConnectivitySettings))]
+    public class SetAzureHDInsightGatewayCredentialCommand : HDInsightCmdletBase
+    {
+        private const string SetByNameParameterSet = "SetByNameParameterSet";
+        private const string SetByResourceIdParameterSet = "SetByResourceIdParameterSet";
+        private const string SetByInputObjectParameterSet = "SetByInputObjectParameterSet";
+
+        #region Input Parameter Definitions
+
+        [Alias("ClusterName")]
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = SetByNameParameterSet,
+            HelpMessage = "Gets or sets the name of the cluster.")]
+        [ResourceNameCompleter("Microsoft.HDInsight/clusters", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        public string Name
+        {
+            get { return _clusterName; }
+            set { _clusterName = value; }
+        }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            ParameterSetName = SetByInputObjectParameterSet,
+            HelpMessage = "Gets or sets the input object.")]
+        [ValidateNotNull]
+        public AzureHDInsightCluster InputObject;
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = SetByResourceIdParameterSet,
+            HelpMessage = "Gets or sets the resource id.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId;
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Gets or sets the login for the cluster's user.")]
+        public PSCredential HttpCredential
+        {
+            get
+            {
+                return _credential == null ? null : new PSCredential(_credential.Username, _credential.Password.ConvertToSecureString());
+            }
+            set
+            {
+                _credential = new BasicAuthenticationCloudCredentials
+                {
+                    Username = value.UserName,
+                    Password = value.Password.ConvertToString()
+                };
+            }
+        }
+
+        [Parameter(HelpMessage = "Gets or sets the name of the resource group.")]
```

Comment:
This aramater is only in the SetByName parameter set, correct?  Please update the parameter appropriately - I believe it is mandatory for this aprameter set